### PR TITLE
Integrate 2023 FERC1 Update

### PR DESCRIPTION
**Pull request by cmgosnell**
# Overview

Working on #3700.
Most of the transform changes are documented over in this scoping issue #3698

2024/07/12 Update: Right now there is one [integration test failure](https://github.com/catalyst-cooperative/pudl/pull/3710#issuecomment-2225592825) coming from `ferc_xbrl_updates`, which @zschira is tackling. Locally I've run the full and fast ETL. With a full db I've run the full validation tests. (I haven't been able to run the `build-deploy-pudl` because the integration test fails.. I'm xfail-ing the one test locally to test everything in one go).

# Testing

How did you make sure this worked? How can a reviewer verify this?

```[tasklist]
# To-do list
- [x] If updating analyses or data processing functions: make sure to update or write data validation tests (e.g.,  `test_minmax_rows()`)
- [x] Update the [release notes](../docs/release_notes.rst): reference the PR and related issues.
- [ ] Ensure docs build, unit & integration tests, and test coverage pass locally with `make pytest-coverage` (otherwise the merge queue may reject your PR)
- [x] Review the PR yourself and call out any questions or issues you have
- [ ] For minor ETL changes or data additions, once `make pytest-coverage` passes, make sure you have a fresh full PUDL DB downloaded locally, materialize new/changed assets and all their downstream assets and [run relevant data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation) using `pytest` and `--live-dbs`.
- [ ] For significant ETL, data coverage or analysis changes, once `make pytest-coverage` passes, ensure the full ETL runs locally and [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation) using `make pytest-validate` (a ~10 hour run). If you can't run this locally, run the `build-deploy-pudl` GitHub Action (or ask someone with permissions to). Then, check the logs on the `#pudl-deployments` Slack channel or `gs://builds.catalyst.coop`.
```

### CODE REVIEW

File: src/pudl/io_managers.py [hunk 1]
```diff
@@ -759,7 +759,7 @@ def __compare_dedupe_methodologies(
 
             # 2024-04-10: this threshold set by looking at existing values for FERC
             # <=2022.
-            threshold_pct = 0.3
+            threshold_pct = 0.32
```
Comment (cmgosnell) [inline src/pudl/io_managers.py:762]: this is a hack! we need to investigate this to see if its really a problem

Comment (cmgosnell) [reply src/pudl/io_managers.py:762]: After some investigation, I do not think this is a problem. For the table that caused this value error in the first place (`electric_plant_in_service_204_duration`), there are 33 excess non-null records out of a possible 10472. Which means that if there were 3 fewer non-null records this test would pass fine np.

After reviewing some of the 2023 excess non-nulls with the apply_diffs methodology, its pretty ambiguous what was supposed to be a record that was updating a previously non-null record into a null vs just that they didn't report anything when they were updating other values. Honestly the fluxxuation in the non-null values from submission to submission was slightly alarming (not all of the values changed by any means! but some were just super different from submission to submission...).

Comment (cmgosnell) [reply src/pudl/io_managers.py]: I'm bumping this up again for the fast tests. This is the same # of excess non-nulls, but with one less year involved.

```
ValueError: Found more than expected excess non-null values using the currently  implemented apply_diffs methodology (#7285) as compared to the best_snapshot methodology (#7254). We expected the apply_diffs methodology to result in no more than 100.32% non-null records but found 100.43%.
```

File: src/pudl/io_managers.py
Comment (cmgosnell) [inline src/pudl/io_managers.py:1]: see [this thread](https://github.com/catalyst-cooperative/pudl/pull/3701#discussion_r1674758622) for what's going on in here

File: src/pudl/package_data/settings/etl_fast.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ description: >
 version: 0.1.0
 datasets:
   ferc1:
-    years: [2020, 2021, 2022]
+    years: [2020, 2021, 2023]
```
Comment (zaneselvans) [inline src/pudl/package_data/settings/etl_fast.yml:38]: Do we expect having a gap in the test years here to be totally fine?

File: src/pudl/transform/ferc1.py [hunk 3]
```diff
@@ -3406,8 +3406,12 @@ class HydroelectricPlantsTableTransformer(Ferc1AbstractTableTransformer):
     table_id: TableIdFerc1 = TableIdFerc1.HYDROELECTRIC_PLANTS
 
     def transform_main(self, df):
-        """Add bespoke removal of duplicate record after standard transform_main."""
-        return super().transform_main(df).pipe(self.targeted_drop_duplicates)
+        """Standard transform_main, bespoke remove duplicate record & remove ``.`` from project_num column."""
+        df = super().transform_main(df).pipe(self.targeted_drop_duplicates)
+        # project_num is an integer column but in 2023 some of them have .'s
+        # as prefixes
+        df.project_num = df.project_num.str.removeprefix(".")
```
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:3413]: At this point it'll be a string column and you're relying on `enforce_schema()` to turn it back into an integer?

**Review by zaneselvans — APPROVED** [review]
It sounds like the missing new-table metadata issue appears in the branch this is merging into so this looks good to go. I asked a couple of non-blocking questions.

File: src/pudl/transform/ferc1.py (hunk 3 above)
Comment (cmgosnell) [reply src/pudl/transform/ferc1.py:3413]: exactly

**Review by zaneselvans — APPROVED** [review]

**Pull request merged**